libbeat: global init state prevents multiple instances of filebeat receiver

While looking into testcases for the new beats receivers I encountered an issue that happens when creating multiple filebeat receivers via a factory:

```
panic: plugin parse_aws_vpc_flow_log registration fail parse_aws_vpc_flow_log exists already [recovered]
	panic: plugin parse_aws_vpc_flow_log registration fail parse_aws_vpc_flow_log exists already
```

Here is a test that reproduces this issue:

<details>
<summary>TestInstantiateReceiverFromFactory</summary>

```go
func TestReceiverFactory(t *testing.T) {
	factory := NewFactory()

	cfg := Config{
		Beatconfig: map[string]interface{}{
			"filebeat": map[string]interface{}{
				"inputs": []map[string]interface{}{
					{
						"type":    "benchmark",
						"enabled": true,
						"message": "test",
						"count":   1,
					},
				},
			},
			"output": map[string]interface{}{
				"otelconsumer": map[string]interface{}{},
			},
			"logging": map[string]interface{}{
				"level": "debug",
				"selectors": []string{
					"*",
				},
			},
			"path.home": t.TempDir(),
		},
	}

	for i := 0; i < 5; i++ {
                // For collector packages < v0.112.0 this is `factory.CreateLogsReceiver`
		receiver, err := factory.CreateLogs(nil, receivertest.NewNopSettings(), &cfg, nil)
		assert.NoError(t, err)
		assert.NotNil(t, receiver)
		assert.NoError(t, receiver.Start(context.Background(), componenttest.NewNopHost()))
		assert.NoError(t, receiver.Shutdown(context.Background()))
	}
}
```

The test output:

```
=== RUN   TestReceiverFactory
--- FAIL: TestReceiverFactory (0.02s)
panic: plugin parse_aws_vpc_flow_log registration fail parse_aws_vpc_flow_log exists already [recovered]
	panic: plugin parse_aws_vpc_flow_log registration fail parse_aws_vpc_flow_log exists already

goroutine 67 [running]:
testing.tRunner.func1.2({0x73e27c0, 0xc0020898e0})
	/home/mauri870/git/go/src/testing/testing.go:1706 +0x21c
testing.tRunner.func1()
	/home/mauri870/git/go/src/testing/testing.go:1709 +0x35e
panic({0x73e27c0?, 0xc0020898e0?})
	/home/mauri870/git/go/src/runtime/panic.go:787 +0x132
github.com/elastic/beats/v7/libbeat/processors.RegisterPlugin({0x7c5b471, 0x16}, 0x7e991e8)
	/home/mauri870/git/elastic/beats/libbeat/processors/registry.go:60 +0x125
github.com/elastic/beats/v7/x-pack/filebeat/processors/aws_vpcflow.InitializeModule()
	/home/mauri870/git/elastic/beats/x-pack/filebeat/processors/aws_vpcflow/parse_aws_vpc_flow_log.go:29 +0x26
github.com/elastic/beats/v7/x-pack/filebeat/include.InitializeModule()
	/home/mauri870/git/elastic/beats/x-pack/filebeat/include/list.go:64 +0x18
github.com/elastic/beats/v7/libbeat/cmd/instance.NewBeat({0x7c184d7, 0x8}, {0x0, 0x0}, {0x0, 0x0}, 0x1, {0xc0010055c0, 0x4, 0x6})
	/home/mauri870/git/elastic/beats/libbeat/cmd/instance/beat.go:245 +0x72
github.com/elastic/beats/v7/libbeat/cmd/instance.NewBeatReceiver({{0x7c184d7, 0x8}, {0x0, 0x0}, {0x0, 0x0}, 0x1, 0x1, {{0x0, 0x0}, ...}, ...}, ...)
	/home/mauri870/git/elastic/beats/libbeat/cmd/instance/beat.go:292 +0xbe
github.com/elastic/beats/v7/x-pack/filebeat/fbreceiver.createReceiver({0xc0011aff20?, 0x7e982b0?}, {{{{0x7c0f18c, 0x3}}, {0xc001315320, 0x24}}, {0xc000aefe80, {0x84d9f30, 0xc0011aff00}, {0x84d9f08, ...}, ...}, ...}, ...)
	/home/mauri870/git/elastic/beats/x-pack/filebeat/fbreceiver/factory.go:45 +0x50e
go.opentelemetry.io/collector/receiver.CreateLogsFunc.CreateLogs(...)
	/home/mauri870/go/pkg/mod/go.opentelemetry.io/collector/receiver@v0.112.0/receiver.go:133
github.com/elastic/beats/v7/x-pack/filebeat/fbreceiver.TestReceiverFactory(0xc001302fc0)
	/home/mauri870/git/elastic/beats/x-pack/filebeat/fbreceiver/receiver_test.go:289 +0x6d9
testing.tRunner(0xc001302fc0, 0x7e97870)
	/home/mauri870/git/go/src/testing/testing.go:1764 +0xf4
created by testing.(*T).Run in goroutine 1
	/home/mauri870/git/go/src/testing/testing.go:1823 +0x409
FAIL	github.com/elastic/beats/v7/x-pack/filebeat/fbreceiver	0.052s
```
</details>

According to @leehinman this is likely a global state issue in libbeat

https://github.com/elastic/beats/blob/6ae50388e8ffa5d74f7f2984ee36b3d93f768787/libbeat/processors/registry.go#L40-L53


Related to https://github.com/elastic/beats/issues/40110.

--- a/CHANGELOG.next.asciidoc
+++ b/CHANGELOG.next.asciidoc
@@ -116,6 +116,7 @@ https://github.com/elastic/beats/compare/v8.8.1\...main[Check the HEAD diff]
 - Add `translate_ldap_attribute` processor. {pull}41472[41472]
 - Remove unnecessary debug logs during idle connection teardown {issue}40824[40824]
 - Fix incorrect cloud provider identification in add_cloud_metadata processor using provider priority mechanism {pull}41636[41636]
+- Prevent panic if libbeat processors are loaded more than once. {issue}41475[41475] {pull}41857[51857]
 
 *Auditbeat*
 

--- a/libbeat/processors/namespace.go
+++ b/libbeat/processors/namespace.go
@@ -38,6 +38,11 @@ type pluginer interface {
 	Plugin() Constructor
 }
 
+var (
+	ErrExistsAlready           error = errors.New("exists already")
+	ErrPluginAlreadyRegistered error = errors.New("non-namespace plugin already registered")
+)
+
 func NewNamespace() *Namespace {
 	return &Namespace{
 		reg: map[string]pluginer{},
@@ -47,7 +52,7 @@ func NewNamespace() *Namespace {
 func (ns *Namespace) Register(name string, factory Constructor) error {
 	p := plugin{NewConditional(factory)}
 	names := strings.Split(name, ".")
-	if err := ns.add(names, p); err != nil {
+	if err := ns.add(names, p); err != nil && !errors.Is(err, ErrExistsAlready) && !errors.Is(err, ErrPluginAlreadyRegistered) {
 		return fmt.Errorf("plugin %s registration fail %w", name, err)
 	}
 	return nil
@@ -59,7 +64,7 @@ func (ns *Namespace) add(names []string, p pluginer) error {
 	// register plugin if intermediate node in path being processed
 	if len(names) == 1 {
 		if _, found := ns.reg[name]; found {
-			return fmt.Errorf("%v exists already", name)
+			return fmt.Errorf("%s %w", name, ErrExistsAlready)
 		}
 
 		ns.reg[name] = p
@@ -71,7 +76,7 @@ func (ns *Namespace) add(names []string, p pluginer) error {
 	if found {
 		ns, ok := tmp.(*Namespace)
 		if !ok {
-			return errors.New("non-namespace plugin already registered")
+			return ErrPluginAlreadyRegistered
 		}
 		return ns.add(names[1:], p)
 	}

--- a/libbeat/processors/namespace_test.go
+++ b/libbeat/processors/namespace_test.go
@@ -65,7 +65,7 @@ func TestNamespaceRegisterFail(t *testing.T) {
 	fatalError(t, err)
 
 	err = ns.Register("test", newTestFilterRule)
-	assert.Error(t, err)
+	assert.NoError(t, err)
 }
 
 func TestNamespaceError(t *testing.T) {

